src/middlewared/middlewared/plugins/service.py

```diff
@@ -639,6 +635,14 @@ def _started_notify(self, verb, what):
             res = True
         return res
 
+    async def _reload_activedirectory(self, **kwargs):
```

Comment:
We already have a _restart_active directory method that calls "/etc/directoryservice/ActiveDirectory/ctl restart"
What I would ideally prefer here is regenerating smb4.conf (if needed) and killing / restarting winbind process as an "AD reload". I.e. _reload_cifs, then restart winbind. This will force new DC connection, but not kill existing SMB sessions.